libs/src/services/solanaWeb3Manager/rewardsAttester.ts

```diff
@@ -729,20 +741,30 @@ export class RewardsAttester {
       wallet,
       completedBlocknumber,
       error,
-      phase
+      phase,
+      nodesToReselect
     }
   }
 
   async _selectDiscoveryNodes() {
     await this._updatePhase('SELECTING_NODES')
-    this.logger.info('Selecting discovery nodes...')
+    this.logger.info(
+      `Selecting discovery nodes with blocklist ${JSON.stringify(
+        this.discoveryNodeBlocklist
+      )}`
+    )
     const startTime = Date.now()
-    const endpoints = await this.libs.discoveryProvider.serviceSelector.findAll(
-      {
+    let endpoints: string[] =
+      (await this.libs.discoveryProvider.serviceSelector.findAll({
         verbose: true,
         whitelist: this.endpointPool.size > 0 ? this.endpointPool : null
-      }
+      })) ?? []
+
+    // Filter out blocklisted nodes
```

Comment:
Can't we send these into `findAll`?

Reply:
I don't think so - findAll doesn't accept a blocklist, only allow

Reply:
sad :'(